web/src/lib/components/Countdown.svelte

```diff
@@ -21,21 +21,24 @@
     }
 
     function refresh() {
-        if (!untilDate) {
+        if (totalSeconds === null) {
             return;
         }
-        if (new Date() > untilDate) {
-            days = hours = minutes = seconds = 0;
-        } else {
-            var delta = Math.abs(untilDate.valueOf() - new Date().valueOf()) / 1000;
-            days = Math.floor(delta / 86400);
-            delta -= days * 86400;
-            hours = Math.floor(delta / 3600) % 24;
-            delta -= hours * 3600;
-            minutes = Math.floor(delta / 60) % 60;
-            delta -= minutes * 60;
-            seconds = delta % 60;
+
+        totalSeconds--;
+
+        if (totalSeconds < 0) {
+            totalSeconds = 0;
         }
```

Comment:
@ibz Maybe you could also put the rest of the code in that function inside this if, so when the counter == 0, you don't do extra calculations, right? If we're at zero, don't calculate anything.

This could also be further improved by calling `refresh()` inside refresh, so when totalSeconds is no longer > 0, you stop calling `refresh()`.

Reply:
Good point. I did all that. I would be afraid to completely stop calling refresh on 0 though, because it could happen that after a future call to the API we would get a different value (> 0) and would still need to do refresh later on...